src/garage/tf/core/parameter.py

```diff
@@ -38,7 +40,11 @@ def parameter(input_var,
             initializer=initializer,
             trainable=trainable)
 
-        broadcast_shape = tf.concat(
-            axis=0, values=[tf.shape(input_var)[:-1], [length]])
+        if pixel_env:
+            batch_dim = [tf.shape(input_var)[0]]
+        else:
+            batch_dim = tf.shape(input_var)[:-1]
```

Comment:
Can you please explain why this is necessary? are you automatically flattening 2D inputs?

Reply:
No, I'm not flattening it. Like I explained in the earlier comment, if we have an input_var of shape `(100, 10, 10, 3)` that is 100 images of shape `(10, 10, 3)`, and my variable of shape 
`(1, )` needs to be `(100, 1)`. I'm picking the batch_dim from the input shape and using that to broadcast the variable. For recurrent policies it needs to be `input_shape[:-1]`